x/wasm/types/proposal.go

```diff
@@ -208,6 +208,10 @@ func (p InstantiateContractProposal) ValidateBasic() error {
 	if p.CodeID == 0 {
 		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "code id is required")
 	}
+	//this code can be removed based on the preference between silent trimming the label or validating the label
+	if p.Label != strings.TrimSpace(p.Label) {
```

Comment:
Good idea to always trim for gov proposals. So that we are safe to not reject any in flight proposals. The code can be simplified though: 
```suggestion
	p.Label != strings.TrimSpace(p.Label)
```

Reply:
do u mean
p.Label = strings.TrimSpace(p.Label)